emacs/radian.el

```diff
@@ -539,11 +567,19 @@ binding the variable dynamically over the entire init-file."
 ;; https://github.com/jwiegley/use-package#notes-about-lazy-loading.
 (setq use-package-always-defer t)
 
+(defmacro radian-use-package (name &rest args)
+  "Like `use-package', but enabled only if the package is not in
+`radian-exclude-packages'. NAME and ARGS are as in `use-package'."
+  (declare (indent defun))
+  `(when (radian-enabled-p ,name)
+     (use-package ,name
+       ,@args)))
```

Comment:
This needs to be a bit more advanced in order to avoid messing up the lockfile. `straight.el` needs to be aware of all the packages that could be installed, even if not all of them are actually installed. What you want to do is call `straight-register-package` unconditionally, and then run the `when` => `use-package` code that you have here.

That's what I do in the current code for `zerodark-theme` (although the `straight-register-package` and `use-package` are in different parts of the file in that case).

Reply:
OK, I tried to do this. I made it so that `straight-register-package` only when the package is disabled. I think it is unnecessary otherwise, correct?

Reply:
Yes, that's right---it doesn't cause any harm to invoke multiple times, but you only _need_ to do it when the package is disabled.